python/ray/_private/structured_logging/formatters.py

```diff
@@ -0,0 +1,27 @@
+import logging
+import json
+
+# A dict_keys object containing the standard attributes of a LogRecord. This is used to
+# help us determine which attributes constitute Ray or user-provided context.
+LOGRECORD_STANDARD_ATTRS = logging.makeLogRecord({"message": "test"}).__dict__.keys()
```

Comment:
I feel this is a bit dangerous to dynamically generate them. Might be better to just list them out explicitly. But also feel it might not be needed, see the comment below.

Reply:
Here is one example that explicitly lists the blacklist: https://github.com/madzak/python-json-logger/blob/5f85723f4693c7289724fdcda84cfc0b62da74d4/src/pythonjsonlogger/jsonlogger.py#L19C1-L19C15. I will try to find whether other projects do that as well. If so, I will list them explicitly.

Reply:
I think it is OK to list all LogRecord attributes explicitly. The LogRecord only adds 2 new attributes in the past 14 years (Python 3.1 and Python 3.12).

https://docs.python.org/3/library/logging.html#logrecord-attributes

Reply:
Update: https://github.com/ray-project/ray/pull/45098/commits/21497b947a64fe4174d48a9e7e0c673d45903918

Reply:
cc @jjyao I decided to list all LogRecord attributes explicitly because of the two reasons mentioned above.